src/awkward/contents/content.py

```diff
@@ -1573,6 +1573,9 @@ def _completely_flatten(
     ) -> list:
         raise ak._util.error(NotImplementedError)
 
+    def drop_none(self):
+        return self._drop_none()
+
```

Comment:
@ioanaif would you mind adding a fallback for the private method? Eventually I'll get around to more typing work and it will be useful to have it :)
```suggestion

def _drop_none(self) -> Content:
    raise ak._errors.wrap_error(NotImplementedError)
```